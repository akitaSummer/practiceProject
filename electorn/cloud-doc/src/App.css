.icon-button {
  background: none;
  border: none;
  color: inherit;
}

.search-height {
  height: calc(1.5em + .75rem + 2px);
  line-height: calc(1.5em + .75rem + 2px);
}

.c-link {
  cursor: pointer;
}

.start-page {
  font-size: 30px;
  height: 300px;
  line-height: 300px;
  text-align: center;
  color: #ccc;
}

.left-panel {
  height: 100vh;
  position: relative;
}

.button-group {
  position: absolute;
  bottom: 0;
  width: 100%;
}

.no-border {
  border-radius: 0 !important;
}
