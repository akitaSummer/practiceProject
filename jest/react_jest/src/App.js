import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import TodoList from './container/TodoList/TodoList'
import NotFoundPage from "./container/NotFoundPage/NotFoundPage";

function App() {
  return (
      <div className="app-container" title='dell lee' data-test='container'>
        <BrowserRouter>
          <Switch>
            <Route path='/' exact component={TodoList}/>
            <Route component={NotFoundPage}/>
          </Switch>
        </BrowserRouter>
      </div>
  );
}

export default App;
