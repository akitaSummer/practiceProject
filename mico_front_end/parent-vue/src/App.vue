<template>
  <div id="nav">
    <router-link to="/vue">child-vue</router-link>
    <div id="vue"></div>
  </div>
  <router-view/>
</template>

<style>

</style>
