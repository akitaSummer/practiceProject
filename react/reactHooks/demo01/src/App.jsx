import React, { useState, useContext } from 'react'

// Context组件：可以将数据传递给后代组件
// 1.初始化Context组件
const CountContext = React.createContext(0)

function A() {
    const [count, setCount] = useState(1)
    // 2.使用Context.Provider组件将需要被传递数据的子组件包裹
    return (
        <>
            <CountContext.Provider value={count}>
                <B></B>
            </CountContext.Provider>
            <button
                onClick={() => {
                    setCount(count + 1)
                }}
            ></button>
        </>
    )
}

function B() {
    return <C></C>
}

function C() {
    const theme = useContext(CountContext)
    return <div>count = {theme}</div>
}

export default A
