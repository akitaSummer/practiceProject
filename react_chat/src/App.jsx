import React from "react";
import PropTypes from 'prop-types'
import { HashRouter, Switch, Route, Redirect } from 'react-router-dom'
import ChatList from './pages/ChatList/ChatList'
import Chat from './pages/Chat/Chat'

const App = (props) => {
  return (
    <>
      <HashRouter>
        <Switch>
          <Route path='/chatlist' render={() => <ChatList users={props.users}/>}/>
          <Route path='/chat/:name' component={<Chat/>}/>
          <Redirect from='/' to='/chatlist' />
        </Switch>
      </HashRouter>
    </>
  )
}

App.propTypes = {
  users: PropTypes.array
}

export default App
