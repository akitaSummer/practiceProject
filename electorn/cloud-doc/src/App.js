import React, { useState } from 'react';
import './App.css';
import { faPlus, faFileImport, faSave } from '@fortawesome/free-solid-svg-icons'
import 'bootstrap/dist/css/bootstrap.min.css'
import FileSearch from "./components/FileSearch";
import FileList from "./components/FileList";
import BottomBtn from "./components/BottomBtn";
import TabList from "./components/TabList";
import Loader from "./components/Loader";
import SimpleMDE from 'react-simplemde-editor'
import 'easymde/dist/easymde.min.css'
import uuidv4 from 'uuid/v4'
import { flattenArr, objToArr, timestampToString } from './utils/helper'
import fileHelper from "./utils/fileHelper";
import useIpcrenderer from "./hooks/useIpcRenderer";

const { join, basename, extname, dirname } = window.require('path')
const { remote, ipcRenderer } = window.require('electron')
const Store = window.require('electron-store')
const fileStore = new Store({ 'name': 'Files Data' })
const settingsStore = new Store({ 'name': 'settings' })

const qiniuConfig = () => ['#accessKey', '#secretKey', "#bucketName", 'enableAutoSync'].every(item => !!settingsStore.get(item))

const saveFilesToStore = (files) => {
  const filesStoreObj = objToArr(files).reduce((result, file) => {
    const { id, title, path, createdAt, isSynced, updatedAt } = file
    result[id] = {
      id,
      title,
      path,
      createdAt,
      isSynced,
      updatedAt
    }
    return result
  }, {})
  fileStore.set('files', filesStoreObj)
}

function App() {

  const [ files, setFiles ] = useState(fileStore.get('files') || {})
  const [ activeFileID, setActiveFileID ] = useState('')
  const [ openFilesIDs, setOpenFileIDs ] = useState([])
  const [ unsaveFilesIDs, setUnsaveFilesIDs ] = useState([])
  const [ searchFiles, setSearchFiles ] = useState([])
  const [ searchEvent, setSearchEvent ] = useState(false)
  const [ isLoading, setIsLoading ] = useState(false)
  const filesArr = objToArr(files)
  const savedLocation = settingsStore.get('saveLocation') || remote.app.getPath('documents')

  const showFiles = searchFiles.length > 0 ? searchFiles : filesArr
  const activeFile = files[activeFileID]
  const openedFiles = openFilesIDs.map(openID => {
    return files[openID]
  })

  const noFile = (id) => {
    window.alert('文件搜索不到')
    const newFiles = { ...files }
    delete newFiles[id]
    saveFilesToStore(newFiles)
    setFiles(newFiles)
  }

  const updateNewFiles = (id, key, value) => {
    const newFile = {
      ...files[id],
      [key]: value
    }
    return { ...files, [id]: newFile}
  }

  const fileClick = (id) => {
    setActiveFileID(id)
    const currentFile = files[id]
    const { title, path, isLoaded } = currentFile
    if (!isLoaded) {
      if (settingsStore.get('enableAutoSync')) {
        ipcRenderer.send('download-file', { key: `${title}.md`, id, path})
      } else {
        fileHelper.readFile(currentFile.path).then((value) => {
          const newFile = {
            ...currentFile,
            isLoaded: true,
            body: value
          }
          setFiles({ ...files, [id]: newFile })
        }).catch(() => {
          noFile(id)
        })
      }
    }
    if (!openFilesIDs.includes(id)) {
      setOpenFileIDs([...openFilesIDs, id])
    }
  }

  const tabClick = (id) => {
    setActiveFileID(id)
  }

  const tabClose = (id) => {
    const tabWithout = openFilesIDs.filter(openID => {
      return openID !== id
    })
    setOpenFileIDs(tabWithout)
    if (tabWithout.length > 0) {
      setActiveFileID(tabWithout[tabWithout.length - 1])
    } else {
      setActiveFileID('')
    }
  }

  const fileChange = (id, value) => {
    if (value !== files[id].body) {
      if (!unsaveFilesIDs.includes(id)) {
        setUnsaveFilesIDs([...unsaveFilesIDs, id])
      }
      const newFiles = updateNewFiles(id, 'body', value)
      setFiles(newFiles)
    }
  }

  const fileDelete = (id) => {
    if (!files[id].path) {
      const { [id]: value, ...afterDelete } = files
      setFiles(afterDelete)
    } else {
      fileHelper.deleteFile(files[id].path).then(() => {
        const newFiles = {...files}
        delete newFiles[id]
        setFiles(newFiles)
        tabClose(id)
      }).catch(() => {
        noFile(id)
      })
    }
  }

  const saveEdit = (id, title, isNew) => {
    const newPath = isNew ? join(savedLocation, `${title}.md`) : join(dirname(files[id].path), `${title}.md`)
    const modifiedFile = { ...files[id], path: newPath, isNew: false, title }
    const newFiles = { ...files, [id]: modifiedFile }
    if (isNew) {
      fileHelper.writeFile(newPath, files[id].body).then(() => {
        setFiles(newFiles)
        saveFilesToStore(newFiles)
      }).catch(() => {
        noFile(id)
      })
    } else {
      fileHelper.renameFile(files[id].path, newPath).then(() => {
        setFiles(newFiles)
        saveFilesToStore(newFiles)
      }).catch(() => {
        noFile(id)
      })
    }
  }

  const saveCurrentFile = () => {
    const { path, body, title } = activeFile
    fileHelper.writeFile(path, body).then(() => {
      setUnsaveFilesIDs(unsaveFilesIDs.filter(id => id !== activeFile.id))
      if (qiniuConfig) {
        ipcRenderer.send('upload-file', {
          key: `${title}.md`,
          path
        })
      }
    })
  }

  const fileSearch = (keyword) => {
    const filesArr = objToArr(files)
    const newFilesArr = filesArr.filter((file) => {
      return file.title.includes(keyword)
    })
    setSearchFiles(newFilesArr)
  }

  const addFile = () => {
    const id = uuidv4()
    const newFiles = {
      ...files,
      [id]: {
        id,
        title: '',
        body: '##请在这里书写Markdown',
        createAt: new Date().getTime(),
        isNew: true
      }
    }
    setFiles(newFiles)
  }

  const fileImport = () => {
    remote.dialog.showOpenDialog({
      title: '选择导入的 Markdown 文件',
      properties: ['openFile', 'multiSelections'],
      filters: [
        {name: 'Markdown files', extensions: ['md']}
      ]
    }).then(({ canceled, filePaths }) => {
      if (!canceled) {
        const fileFinderPath = filePaths.filter(path => {
          return !Object.values(files).find(file => {
            return file.path === path
          })
        })
        const importFilesArr = fileFinderPath.map(path => {
          return {
            id: uuidv4(),
            title: basename(path, extname(path)),
            path
          }
        })
        const newFiles = {
          ...files,
          ...flattenArr(importFilesArr)
        }
        setFiles(newFiles)
        saveFilesToStore(newFiles)
        if (importFilesArr.length > 0) {
          remote.dialog.showMessageBox({
            type: 'info',
            title: `成功导入了${importFilesArr.length}个文件`,
            message: `成功导入了${importFilesArr.length}个文件`
          })
        }
      }
    })
  }

  const fileSearchEvent = () => {
    setSearchEvent(!searchEvent)
  }

  const activeFileUploaded = () => {
    const { id } = activeFile
    const modifiedFile = { ...files[id], isSynced: true, updatedAt: new Date().getTime()}
    const newFiles = { ...files, [id]: modifiedFile }
    setFiles(newFiles)
    saveFilesToStore(newFiles)
  }

  const activeFileDownloaded = (event, message) => {
    const currentFile = files[message.id]
    const { id, path } = currentFile
    fileHelper.readFile(path).then(value => {
      let newFile
      if (message.status === 'download-success') {
        newFile = { ...files[id], body: value, isLoaded: true, isSynced: true, updatedAt: new Date().getTime() }
      } else {
        newFile = { ...files[id], body: value, isLoaded: true }
      }
      const newFiles = { ...files, [id]: newFile }
      setFiles(newFiles)
      saveFilesToStore(newFiles)
    })
  }

  const filesUploaded = () => {
    const newFiles = objToArr(files).reduce((result, file) => {
      const currentTime = new Date().getTime()
      result[file.id] = {
        ...files[file.id],
        isSynced: true,
        updatedAt: currentTime
      }
    }, {})
    setFiles(newFiles)
    saveFilesToStore(newFiles)
  }

  useIpcrenderer({
    'create-new-file': addFile,
    'import-file': fileImport,
    'save-edit-file': saveCurrentFile,
    'search-file': fileSearchEvent,
    'active-file-uploaded': activeFileUploaded,
    'file-downloaded': activeFileDownloaded,
    'loading-status': (message, status) => {
      setIsLoading(status)
    },
    'files-uploaded': filesUploaded
  })

  return (
    <div className="App container-fluid px-0">
      {
        isLoading && <Loader></Loader>
      }
      <div className="row no-gutters">
        <div className="col-3 bg-light left-panel">
          <FileSearch
            title='我的云文档'
            onFileSearch={fileSearch}
            searchEvent={searchEvent}
            onFileSearchEvent={fileSearchEvent}
          ></FileSearch>
          <FileList
            files={showFiles}
            onFileClick={fileClick}
            onFileDelete={fileDelete}
            onSaveEdit={saveEdit}
          ></FileList>
          <div className="row no-gutters button-group no-border">
            <div className="col">
              <BottomBtn
                text='新建'
                colorClass='btn-primary'
                icon={faPlus}
                onBtnClick={addFile}
              ></BottomBtn>
            </div>
            <div className="col">
              <BottomBtn
                text='导入'
                colorClass='btn-success'
                icon={faFileImport}
                onBtnClick={fileImport}
              ></BottomBtn>
            </div>
          </div>
        </div>
        <div className="col-9 right-panel">
          {
            activeFile ?
              <>
                <TabList
                  files={openedFiles}
                  activeId={activeFileID}
                  unsaveIds={unsaveFilesIDs}
                  onTabClick={tabClick}
                  onCloseTab={tabClose}
                ></TabList>
                <SimpleMDE
                  key={activeFile && activeFile.id}
                  value={activeFile && activeFile.body}
                  onChange={(value) => {fileChange(activeFile.id, value)}}
                  options={{
                    minHeight: '515px'
                  }}
                ></SimpleMDE>
                { activeFile.isSynced && <span className="sync-status">已同步，上次同步{timestampToString(activeFile.updatedAt)}</span>}
              </>
              :
              <div className='start-page'>
                选择或者创建新的 Markdown 文档
              </div>
          }

        </div>
      </div>
    </div>
  );
}

export default App;
