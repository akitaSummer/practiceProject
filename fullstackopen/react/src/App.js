import React, { useState } from 'react';
import Filter from "./component/Filter";
import PersonForm from "./component/PersonForm";
import Persons from "./component/Persons";
import './App.css';

const App = () => {

  const [persons, setPersons] = useState([
    { name: 'Arto Hellas', number: '040-123456' },
    { name: 'Ada Lovelace', number: '39-44-5323523' },
    { name: 'Dan Abramov', number: '12-43-234345' },
    { name: 'Mary Poppendieck', number: '39-23-6423122' }
  ])
  const [showPersons, setShowPersons] = useState(persons)


  return (
    <div>
      <h2>Phonebook</h2>

      <Filter persons={persons} setShowPersons={setShowPersons}/>

      <h3>Add a new</h3>

      <PersonForm setPersons={setPersons} persons={persons} setShowPersons={setShowPersons}/>

      <h3>Numbers</h3>

      <Persons showPersons={showPersons}/>
    </div>
)

}

export default App;
