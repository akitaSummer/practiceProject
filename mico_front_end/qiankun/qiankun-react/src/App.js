import logo from "./logo.svg";
import "./App.css";
import { BrowserRouter, Route, Link } from "react-router-dom";

function App() {
  return (
    <BrowserRouter basename='/react'>
      <Link to=''>Home</Link>
      <Link to='/about'>About</Link>
      <Route path='/' exact render={() => (
         <div className='App'>
          <header className='App-header'>
            <img src={logo} className='App-logo' alt='logo' />
            <p>
              Edit <code>src/App.js</code> and save to reload.
            </p>
            <a
              className='App-link'
              href='https://reactjs.org'
              target='_blank'
              rel='noopener noreferrer'>
              Learn React
            </a>
          </header>
        </div> 
      )}/>
      <Route path='/about' exact render={() => <h1>about</h1>}/>
    </BrowserRouter>
  );
}

export default App;
