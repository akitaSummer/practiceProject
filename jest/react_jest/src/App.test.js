import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import App from './App';

// import ReactDOM from 'react-dom';
// import { render } from '@testing-library/react';
//
// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/Hello world/i);
//   expect(linkElement).toBeInTheDocument();
// });
//
// test('renders without crashing', () => {
//   const div = document.createElement('div')
//   ReactDOM.render(<App/>, div)
//   // ReactDOM.unmountComponentAtNode(div)
//   const container = div.getElementsByClassName('App')
//   expect(container.length).toBe(1)
// })

Enzyme.configure({ adapter: new Adapter() })

// test('Enzyme shallow', () => {
//   // shallow是浅复制/浅渲染
//   // 适用于单元测试
//   const wapper_1 = shallow(<App/>)
//   const container_1 = wapper_1.find('[data-test="container"]')
//   // expect(wapper.find('.app-container').length).toBe(1)
//   // console.log(wapper.debug())
//   // expect(wapper.find('.app-container').prop('title')).toBe('dell lee')
//   // expect(wapper.find('[data-test="container"]').length).toBe(1)
//   expect(container_1).toExist() // 存在
//   expect(container_1).toHaveProp('title', 'dell lee') // 存在属性
//   expect(container_1).toMatchSnapshot()
//   // mount是全渲染
//   // 适合集成测试，他会将所有内容渲染
//   const wapper_2 = mount(<App/>)
//   const container_2 = wapper_2.find('[data-test="container"]')
//   expect(container_2).toExist() // 存在
//   expect(container_2).toHaveProp('title', 'dell lee') // 存在属性
// })

