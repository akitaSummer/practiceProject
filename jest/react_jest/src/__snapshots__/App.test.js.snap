// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enzyme shallow 1`] = `
<div
  className="app-container"
  data-test="container"
  title="dell lee"
>
  Hello world
</div>
`;
